sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessPipelineRunner.java

```diff
@@ -173,13 +179,26 @@
 
   ////////////////////////////////////////////////////////////////////////////////////////////////
   private final InProcessPipelineOptions options;
+  private Supplier<ExecutorService> executorServiceSupplier;
+  private Supplier<Clock> clockSupplier;
 
   public static InProcessPipelineRunner fromOptions(PipelineOptions options) {
     return new InProcessPipelineRunner(options.as(InProcessPipelineOptions.class));
   }
 
   private InProcessPipelineRunner(InProcessPipelineOptions options) {
     this.options = options;
+    if (options.getExecutorServiceFactory() == null) {
+      executorServiceSupplier = new FixedThreadPoolSupplier();
```

Comment:
I have to set it somewhere during construction; it seems reasonable to only create the supplier that will be used.